### PR TITLE
Add enter / exit / transfer commands to CLI

### PR DESCRIPTION
**Pull request by maaktweluit**
Add CLI commands, GSB messages and driver implementation stubs for:

```
yagna payment enter --driver=zksync --network=rinkeby
yagna payment exit --driver=zksync --network=rinkeby [optional address]
yagna payment transfer —driver=zksync —network=rinkeby [address]
```

### CODE REVIEW

File: core/model/src/driver.rs [hunk 1]
```diff
@@ -236,3 +236,92 @@ impl RpcMessage for ValidateAllocation {
     type Item = bool;
     type Error = GenericError;
 }
+
+// ************************** ENTER **************************
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Enter {
+    amount: String,
```
Comment (Wiezzel) [inline core/model/src/driver.rs:244]: Parsing amount should be done in CLI, not in driver.
```suggestion
    amount: BigDecimal,
```

File: core/model/src/driver.rs [hunk 2]
```diff
@@ -236,3 +236,92 @@ impl RpcMessage for ValidateAllocation {
...
+    type Item = String; // Transaction Identifier
+    type Error = GenericError;
+}
+
+// ************************** EXIT **************************
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Exit {
+    to: Option<String>,
+    amount: Option<String>,
```
Comment (Wiezzel) [inline core/model/src/driver.rs:270]: Parsing amount should be done in CLI, not in driver.

```suggestion
    amount: Option<BigDecimal>,
```

File: core/model/src/driver.rs [hunk 3]
```diff
@@ -236,3 +236,92 @@ impl RpcMessage for ValidateAllocation {
...
+    network: Option<String>,
+    token: Option<String>,
+}
+
+impl Exit {
+    pub fn new(
+        to: Option<String>,
+        amount: Option<String>,
```
Comment (Wiezzel) [inline core/model/src/driver.rs:278]: Parsing amount should be done in CLI, not in driver.
```suggestion
        amount: Option<BigDecimal>,
```

File: core/model/src/driver.rs [hunk 4]
```diff
@@ -236,3 +236,92 @@ impl RpcMessage for ValidateAllocation {
...
+    type Item = String; // Transaction Identifier
+    type Error = GenericError;
+}
+
+// ************************** TRANSFER **************************
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Transfer {
+    to: String,
+    amount: String,
```
Comment (Wiezzel) [inline core/model/src/driver.rs:302]: Parsing amount should be done in CLI, not in driver.
```suggestion
    amount: BigDecimal,
```

File: core/model/src/driver.rs [hunk 5]
```diff
@@ -236,3 +236,92 @@ impl RpcMessage for ValidateAllocation {
...
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Enter {
+    amount: String,
+    network: Option<String>,
+    token: Option<String>,
+}
+
+impl Enter {
+    pub fn new(amount: String, network: Option<String>, token: Option<String>) -> Enter {
```
Comment (Wiezzel) [inline core/model/src/driver.rs:250]: Parsing amount should be done in CLI, not in driver.
```suggestion
    pub fn new(amount: BigDecimal, network: Option<String>, token: Option<String>) -> Enter {
```

File: core/model/src/driver.rs [hunk 6]
```diff
@@ -236,3 +236,92 @@ impl RpcMessage for ValidateAllocation {
...
+    to: String,
+    amount: String,
+    network: Option<String>,
+    token: Option<String>,
+}
+
+impl Transfer {
+    pub fn new(
+        to: String,
+        amount: String,
```
Comment (Wiezzel) [inline core/model/src/driver.rs:310]: Parsing amount should be done in CLI, not in driver.
```suggestion
        amount: BigDecimal,
```

File: core/payment/src/wallet.rs [hunk 7]
```diff
@@ -0,0 +1,40 @@
+use ya_core_model::driver::{driver_bus_id, Enter, Exit, Transfer};
+use ya_service_bus::typed as bus;
+
+pub async fn enter(
+    amount: String,
```
Comment (Wiezzel) [inline core/payment/src/wallet.rs:5]: Parsing amount should be done in CLI, not in driver.
```suggestion
    amount: BigDecimal,
```

File: core/payment/src/wallet.rs [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+) -> anyhow::Result<String> {
+    let driver_id = driver_bus_id(driver);
+    let message = Enter::new(amount, network, token);
+    let tx_id = bus::service(driver_id).call(message).await??;
+    Ok(tx_id)
+}
+
+pub async fn exit(
+    to: Option<String>,
+    amount: Option<String>,
```
Comment (Wiezzel) [inline core/payment/src/wallet.rs:18]: Parsing amount should be done in CLI, not in driver.
```suggestion
    amount: Option<BigDecimal>,
```

File: core/payment/src/wallet.rs [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+) -> anyhow::Result<String> {
+    let driver_id = driver_bus_id(driver);
+    let message = Exit::new(to, amount, network, token);
+    let tx_id = bus::service(driver_id).call(message).await??;
+    Ok(tx_id)
+}
+
+pub async fn transfer(
+    to: String,
+    amount: String,
```
Comment (Wiezzel) [inline core/payment/src/wallet.rs:31]: Parsing amount should be done in CLI, not in driver.
```suggestion
    amount: BigDeciaml,
```

**Review by Wiezzel — CHANGES_REQUESTED** [review]
Parsing amount should be done in CLI, not in driver.

**Comment by maaktweluit** [comment]
@Wiezzel Was unsure about amount, since it will be put into a string when send over GSB anyhow...

Will update it to parse earlier :)

**Review by Wiezzel — APPROVED** [review]

**Review by pnowosie — APPROVED** [review]
Great work!

**Pull request merged**